engine/src/main/java/com/datatorrent/stram/plan/logical/LogicalPlan.java

```diff
@@ -2424,4 +2435,103 @@ public int hashCode()
     return result;
   }
 
+  /**
+   * Return list of operators in topological sorted order. The Delay operators
+   * are considered as downstream operator.
+   * @return
+   */
+  @VisibleForTesting
+  protected List<OperatorMeta> getSortedOperators()
```

Comment:
Call this getOperatorsTopographicalOrder() and make it available in the interface, see lifecycle comment below.